### DISCUSSION TITLE
How to Fix - Future partitions not being created

### DISCUSSION THREAD

**Discussion by vinkaga** [pgpartman/pg_partman#439] [Q&A] [answered]
Hello,
I created weekly partitions using this documentation https://github.com/pgpartman/pg_partman/blob/master/doc/pg_partman_howto_native.md. It created partitions and the performance looked good.

But since then, there are no new weekly partitions been created (should have been at least 2 more base on ISO week convention). My questions are
1. Is the new data in `_default` partition (which doesn't have a corresponding partition)?
2. What did I miss that prevented creating future partitions?
3. How do I create the missing partitions and future partitions?
4. How do I move the data into the partitions (if it is in `_default`)?

**Comment by keithf4** [comment, marked as answer]
By default, no new partitions are created by pg_partman unless new data has been added to the partition set. All future partition creation is managed by calling the run_maintenance() function on a regular basis. New child tables are not automatically created simply based on incoming data to avoid contention issues. You can either call run maintenance via cron or you can use the built in background worker (see documentation).

If you had added new data, and there are appropriate child tables for that data, it should go to those child tables. If there is no matching child table for that data, then it goes to the default. You will then have to use the partition_data_*() functions/procedures to move that data out to the proper child tables.

**Comment by vinkaga** [comment]
@keithf4, What is the preferred way to execute `run_maintenance()` on a regular basis? Should it be a chron job?

**Comment by vinkaga** [comment]
Never mind, that was a dumb question :-). I'll run it in a lambda.
